### PR TITLE
refactor: add loadbalancer coder for vlan and vxlan

### PR DESCRIPTION
**Pull request by zhang-shengping**

### CODE REVIEW

**Review by zhaoqin-github — CHANGES_REQUESTED** [review]
Please follow ResourceManager's style to implement

File: f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py [hunk 1]
```diff
@@ -167,26 +168,102 @@ def __init__(self, driver):
...
+                                                         self.driver)
         self.mutable_props = {
             "name": "description",
             "description": "description",
             "admin_state_up": "enabled"
         }
+        self.all_subnet_hints = {}
 
-    def _create_payload(self, loadbalancer, service):
+    def _create_vip(self, loadbalancer, service):
```
Comment (zhaoqin-github) [inline f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py]: Please do not modify this name

File: f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py [hunk 2]
```diff
@@ -167,26 +168,102 @@ def __init__(self, driver):
...
         vip = virtual_address.VirtualAddress(self.driver.service_adapter,
                                              loadbalancer)
-        return vip.model()
+        return vip
 
     @log_helpers.log_method_call
     def create(self, loadbalancer, service, **kwargs):
-        # TODO(qzhao): Future work
-        pass
+        self._pre_create(service)
```
Comment (zhaoqin-github) [inline f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py:194]: I suggest to write it in this way:

```python
def create(self, loadbalancer, service, **kwargs):
    self._pre_create(service)
    super(LoadBalancerManager, self).create(loadbalancer, service)
    self._post_create(service)
```

File: f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py [hunk 3]
```diff
@@ -167,26 +168,102 @@ def __init__(self, driver):
...
+        self._create_lb(service)
+        self._post_create(service)
+
+    def _post_create(self, service):
+        # create fdb for vxlan tunnel
+        if not self.driver.conf.f5_global_routed_mode:
+            self.driver.network_builder.update_bigip_l2(service)
+
+    def _create_lb(self, service):
```
Comment (zhaoqin-github) [inline f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py]: We may not need _create_lb()

File: f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py [hunk 4]
```diff
@@ -167,26 +168,102 @@ def __init__(self, driver):
...
+        bigips = self.driver.get_config_bigips()
+        loadbalancer = service["loadbalancer"]
+        vip = self._create_vip(loadbalancer, service)
+
+        for bigip in bigips:
+            vip.assure(bigip)
+
+        # allow address pair
+        if self.driver.l3_binding:
```
Comment (zhaoqin-github) [inline f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py:207]: If it is something that need to be done after all vip are created, you may move it to _post_create()

**Review by zhaoqin-github — CHANGES_REQUESTED** [review]

File: f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py (hunk 4 above)
Comment (zhang-shengping) [reply f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py:207]: > If it is something that need to be done after all vip are created, you may move it to _post_create()

value of vip_address has already existed. This needs some scenarios to confirm if it's necessary to be created before vip.

File: f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py [hunk 5]
```diff
@@ -187,14 +191,71 @@ def _create_payload(self, loadbalancer, service):
     @serialized('LoadBalancerManager.create')
     @log_helpers.log_method_call
     def create(self, loadbalancer, service, **kwargs):
-        # TODO(qzhao): Future work
-        pass
+        self._pre_create(service)
+        super(LoadBalancerManager, self).create(
```
Comment (zhaoqin-github) [inline f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py:195]: It may have deadlock issue, but we should solve the problem in another commit.

**Review by zhaoqin-github — APPROVED** [review]
Looks fine

File: f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py (hunk 5 above)
Comment (zhang-shengping) [reply f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py:195]: Yes， it has.

**Pull request merged**